### ISSUE TITLE
If the namespace doesn't exist `--dry-run` fails

### ISSUE THREAD

**Issue by doubtingben** [stefanprodan/timoni#65]
If the namespace doesn't exist on a `--dry-run`, the dry run errors.
```
$ timoni -n test apply podinfo oci://ghcr.io/stefanprodan/modules/podinfo --version latest --dry-run  
pulling oci://ghcr.io/stefanprodan/modules/podinfo:latest
using module timoni.sh/podinfo version 6.3.5
ServiceAccount/test/podinfo namespace not specified, error: namespaces "test" not found
Service/test/podinfo namespace not specified, error: namespaces "test" not found
Deployment/test/podinfo namespace not specified, error: namespaces "test" not found
```

Creating the namespace first allows a dry-run, but `--diff` doesn't produce any new output.
```
$ kubectl create ns test
namespace/test created

$ timoni -n test apply podinfo oci://ghcr.io/stefanprodan/modules/podinfo --version latest --dry-run  --diff
pulling oci://ghcr.io/stefanprodan/modules/podinfo:latest
using module timoni.sh/podinfo version 6.3.5
ServiceAccount/test/podinfo created (server dry run)
Service/test/podinfo created (server dry run)
Deployment/test/podinfo created (server dry run)

$ timoni -n test apply podinfo oci://ghcr.io/stefanprodan/modules/podinfo --version latest --dry-run  
pulling oci://ghcr.io/stefanprodan/modules/podinfo:latest
using module timoni.sh/podinfo version 6.3.5
ServiceAccount/test/podinfo created (server dry run)
Service/test/podinfo created (server dry run)
Deployment/test/podinfo created (server dry run)
```

**Comment by stefanprodan** [comment]
Diff only produces an output for changes to existing resources, like kubectl diff does.

**Comment by stefanprodan** [comment]
If the namespace doesn't exists, instead of an error, the output should list all resources as `created (server dry run)`.

**Comment by doubtingben** [comment]
> Diff only produces an output for changes to existing resources, like kubectl diff does.

I'm not sure I understand.
kubectl diff shows resource creations.

```
$ timoni -n test apply podinfo oci://ghcr.io/stefanprodan/modules/podinfo --version latest

$ kubectl get svc podinfo -o yaml > podinfo.yaml

$ kubectl delete svc podinfo

$ timoni -n test apply podinfo oci://ghcr.io/stefanprodan/modules/podinfo --version latest --dry-run  --diff
pulling oci://ghcr.io/stefanprodan/modules/podinfo:latest
using module timoni.sh/podinfo version 6.3.5
ServiceAccount/test/podinfo unchanged (server dry run)
Service/test/podinfo created (server dry run)
Deployment/test/podinfo unchanged (server dry run)

# small edit to yaml
$ kubectl diff --server-side=true -f podinfo.yaml 
diff -u -N /var/folders/59/n8859s6x13g3cn8b09xg13tr0000gn/T/LIVE-3149638589/v1.Service.test.podinfo /var/folders/59/n8859s6x13g3cn8b09xg13tr0000gn/T/MERGED-2622538087/v1.Service.test.podinfo
--- /var/folders/59/n8859s6x13g3cn8b09xg13tr0000gn/T/LIVE-3149638589/v1.Service.test.podinfo    2023-04-10 08:59:41
+++ /var/folders/59/n8859s6x13g3cn8b09xg13tr0000gn/T/MERGED-2622538087/v1.Service.test.podinfo  2023-04-10 08:59:41
@@ -0,0 +1,31 @@
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: "2023-04-10T12:59:41Z"
+  labels:
+    app.kubernetes.io/name: podinfo
+    app.kubernetes.io/version: 6.3.5
+    instance.timoni.sh/name: podinfo
+    instance.timoni.sh/namespace: test
+  name: podinfo
+  namespace: test
+  uid: 4be79e6a-99fd-4ab4-85da-935e4594cc66
+spec:
+  clusterIP: 10.245.0.0
+  clusterIPs:
+  - 10.245.0.0
+  internalTrafficPolicy: Cluster
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: http
+  selector:
+    app.kubernetes.io/name: podinfo
+  sessionAffinity: None
+  type: ClusterIP
+status:
+  loadBalancer: {}
```
